src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/GameFacePixelHashObserver.cs

```diff
@@ -107,7 +107,11 @@ private void UpdateGameFacePixelHash()
                 if (cohtmlViewTexture != null)
                 {
                     var wasActive = Interlocked.CompareExchange(ref _requestInProgress, string.Empty, null);
-                    if (wasActive == null)
+                    
+                    // If we are running in -nographics mode, the async task below fails, causing an exception inside
+                    // the AsyncGPUReadback.Request that is difficult to catch. This ensures that the image data
+                    // is only read when we have graphics
+                    if (wasActive == null && SystemInfo.graphicsDeviceType != GraphicsDeviceType.Null)
```

Comment:
This needs to be closer to the start of the function.  We don't even need to bother with the cohtmlviewtexture lookup if we don't have graphics output